GlobalUtilities.cpp

```diff
@@ -275,50 +275,59 @@ void AVGlobals::GetDefaults(const string &defaultsFile) {
   bShowBody = true;
   startWithValueModel = false;
 
+
   // try to find the defaultsFile
+
   string fullDefaultsFile;
+  const string homePath = getenv("HOME");
+
+  std::vector<string> fullDefaultsFileList;
+#ifdef AMRVIS_CONFIG_FILEPATH
+  string configFilepath(AMRVIS_CONFIG_FILEPATH);
+  fullDefaultsFileList.push_back(configFilepath + "/" + defaultsFile);
+#endif
+  fullDefaultsFileList.push_back("./" + defaultsFile);
+  fullDefaultsFileList.push_back(homePath + "/" + defaultsFile);
+  fullDefaultsFileList.push_back(homePath + "/." + defaultsFile);
 
-  fullDefaultsFile = "./" + defaultsFile;     // try dot first
   ifstream defs;
-  defs.open(fullDefaultsFile.c_str());
 
-  if(defs.fail()) {  // try ~ (tilde)
-    if(ParallelDescriptor::IOProcessor()) {
-      cout << "Cannot find amrvis defaults file:  " << fullDefaultsFile << endl;
-    }
-    fullDefaultsFile  = getenv("HOME");
-    fullDefaultsFile += "/";
-    fullDefaultsFile += defaultsFile;
-    defs.clear();  // must do this to clear the fail bit
-    defs.open(fullDefaultsFile.c_str());
-    if(defs.fail()) {  // try ~/.  (hidden file)
-      if(ParallelDescriptor::IOProcessor()) {
-        cout << "Cannot find amrvis defaults file:  " << fullDefaultsFile << endl;
-      }
-      fullDefaultsFile  = getenv("HOME");
-      fullDefaultsFile += "/.";
-      fullDefaultsFile += defaultsFile;
+  bool fileFound = false;
+
+  for (string fileLoc : fullDefaultsFileList){
```

Comment:
Yes, it's influenced by python.  But string is not a simple type.  So this is better.
```suggestion
  for (string const& fileLoc : fullDefaultsFileList){
```

Reply:
Good to know, made the change!